commit 082bc9f77b200eb48a5f1147163dea9c9d02d44c
Author: Mike Gilbert <floppym@gentoo.org>
Date:   Sat Mar 5 17:30:48 2016 -0500

    10_linux: Fix grouping of tests for GRUB_DEVICE
    
    Commit 7290bb562 causes GRUB_DISABLE_LINUX_UUID to be ignored due to
    mixing of || and && operators. Add some parens to help with that.

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 5a78513..de9044c 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -47,7 +47,7 @@ esac
 # and mounting btrfs requires user space scanning, so force UUID in this case.
 if [ "x${GRUB_DEVICE_UUID}" = "x" ] || [ "x${GRUB_DISABLE_LINUX_UUID}" = "xtrue" ] \
     || ! test -e "/dev/disk/by-uuid/${GRUB_DEVICE_UUID}" \
-    || test -e "${GRUB_DEVICE}" && uses_abstraction "${GRUB_DEVICE}" lvm; then
+    || ( test -e "${GRUB_DEVICE}" && uses_abstraction "${GRUB_DEVICE}" lvm ); then
   LINUX_ROOT_DEVICE=${GRUB_DEVICE}
 else
   LINUX_ROOT_DEVICE=UUID=${GRUB_DEVICE_UUID}
